TombEngine/Objects/TR3/Entity/SophiaLeigh.cpp

```diff
@@ -696,8 +749,8 @@ namespace TEN::Entities::Creatures::TR3
 		}
 		else
 		{
-
-			if (item.TriggerFlags & (int)SophiaOCB::OCB_Tower)
+			if (item.TriggerFlags == (int)SophiaOCB::Tower ||
+				item.TriggerFlags == (int)SophiaOCB::TowerWithVolume)
 				SophiaLeighTowerControl(item, &creature, &data);
 			else
 				SophiaLeighNormalControl(item, &creature, &data);
```

Comment:
The current state works, but I wonder, couldn't we just discard the OCBs? In a way that she activates the Tower Mode if she has the AI_X1 object in her sector.

So she could use NormalControl for default behaviour, but if she has the AI X1 then do the Tower control behaviour. (Because right now, the builder still must do both, put the OCB and place the AI X1 on her initial sector and her path).

For the lua version, it could change the line 326
```c++
if (item.TriggerFlags == (int)SophiaOCB::Tower)
```
For any flag stored in the ItemFlags[] like:
```c++
bool flagFollowLara = ((item->ItemFlags[6] & (1 << 0)) != 0);  //Check the first bit of the ItemFlags[6]
```
```c++
if (flagFollowLara )
{
	// If enemy is above, get to next AI_X1.
	if (ai.verticalDistance >= SOPHIALEIGH_Y_DISTANCE_RANGE)
		creature->LocationAI++;

	// If enemy is below, get to previous AI_X1.
	else if (ai.verticalDistance <= -SOPHIALEIGH_Y_DISTANCE_RANGE)
		creature->LocationAI--;
}
```
In Lua it can be changed the ItemFlags, so it can be easy stop that flag if builder is going to control Sophia manually with a lua module.

In this way, no OCBs would be needed.

Reply:
Clarified along Discord.